flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionVertex.java

```diff
@@ -249,6 +254,19 @@ public CoLocationConstraint getLocationConstraint() {
 		return locationConstraint;
 	}
 
+	public InputSplit getNextInputSplit(int index, String host) {
+		final int taskId = this.getParallelSubtaskIndex();
+		synchronized (this.inputSplits) {
+			if (index < this.inputSplits.size()) {
+				return this.inputSplits.get(index);
```

Comment:
there is no need to try to get input split from the input split history.

Reply:
Thanks, removed.
It was intent for support speculative execution, that multiple version of the same executionVertex can be run at same time, but i will postpone the feature and add it later